frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BasePullBranchAction.java

```diff
@@ -95,4 +104,21 @@ private void doPull(Project project, GitRepository gitRepository, String branchN
     new FetchBackgroundable(project, gitRepository, refspecRemoteLocal, GitRemote.DOT,
         /* taskTitle */ GitMacheteBundle.message("action.GitMachete.BasePullBranchAction.task-title")).queue();
   }
+
+  private void doPull(Project project, GitRepository gitRepository, String branchName) {
+    var firstRemote = Option.ofOptional(gitRepository.getRemotes().stream().findFirst()).getOrNull();
+    if (firstRemote == null) {
+      LOG.error("Selected remote can't be null here.");
```

Comment:
Also this message can be misleading because there is no remote that we selected (at least we selected local branch).